src/Http/ServerRequest.php

```diff
@@ -1410,12 +1422,26 @@ public function query($name = null)
     /**
      * Read a specific query value or dotted path.
      *
-     * @param string $name The name or dotted path to the query param.
-     * @param mixed $default The default value if the named parameter is not set.
+     * Developers are encouraged to use getQueryParams() when possible as it is PSR-7 compliant, and this method
+     * is not. You can future proof your code by using the following example.
```

Comment:
Are we sure we want to get rid of the Hash::get() convenience method here in future versions?
To me it sounds like a good idea to keep `getQuery('Post.id')` as an easier way of `Hash::get($request->getQueryParams(), 'Post.id', null);`.